### PR TITLE
fix(json codec): Fix deserializing non-object values with the `Vector` namespace

### PR DESCRIPTION
**Pull request by fuchsnj**
Fixes the issue described here: https://github.com/vectordotdev/vector/discussions/17889#discussioncomment-6800503

The json codec now supports deserializing non-object JSON values when using the vector namespace.

### CODE REVIEW

File: lib/codecs/src/decoding/format/native_json.rs [hunk 1]
```diff
@@ -132,8 +132,8 @@ mod test {
 
         let events = deserializer.parse(input, LogNamespace::Legacy).unwrap();
 
-        let event1 = Event::try_from(json1).unwrap();
-        let event2 = Event::try_from(json2).unwrap();
+        let event1 = Event::from_json_value(json1, LogNamespace::Legacy).unwrap();
```
Comment (jszwedko) [inline lib/codecs/src/decoding/format/native_json.rs:135]: Unrelated, but this is making me wonder: is data passed through correctly from Vector sink to Vector source when using log namespacing on either or both sides?

**Review by jszwedko — APPROVED** [review]
🎉

**Review by pront — APPROVED** [review]
Changes look good and test coverage sufficient.

File: lib/codecs/src/decoding/format/native_json.rs (hunk 1 above)
Comment (fuchsnj) [reply lib/codecs/src/decoding/format/native_json.rs:135]: The Vector sink/source uses protobuf to send the events. The protobuf encodes the raw `Value` for the event and metadata, so it should keep it in the correct format regardless of the log namespacing used. (It used to store only only a map for the event, assuming it was an object, but that was changed a while ago).

**Pull request merged**